migrate.sh

```diff
@@ -35,30 +54,63 @@ function urlencode() {
   echo "$e"
 }
 
+function getObjects() {
+  local type=${1-}
+  if [[ "$type" == "archived" ]]; then
+    type="&archived=true"
+  fi
+
+  local headerUrl
+  local pages
+  headerUrl=$(curl "${CURL_PARAMS}" -sS --head --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}${type}")
+  pages=$(grep -oP '(?<=x-total-pages: ).*' <<< "${headerUrl}" | tr -d '\r')
+
+  for ((i=1; i<="${pages}"; i++)); do
+    local -a objects
+    mapfile -t objects < <(curl "${CURL_PARAMS}" -sS --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}${type}&page=${i}" | jq -r '.[].path_with_namespace')
+    local object
+    for object in "${objects[@]}"; do
+      echo "${object}"
+    done
+  done
+}
+
 function migrateGroup() {
   local groupPath=$1
   echo "Migrating group '${groupPath}'"
 
-  migrateGroupVariables "${groupPath}"
-  migrateBadges "${groupPath}" "groups"
+  if [[ "$MIGRATE_GROUP_VARIABLES" == "yes"  ]]; then
+    migrateGroupVariables "${groupPath}"
+  fi
+
+  if [[ "$MIGRATE_BADGES" == "yes"  ]]; then
+    migrateBadges "${groupPath}" "groups"
+  fi
 
   local groupPathEncoded
   groupPathEncoded=$(urlencode "${groupPath}")
   local groupsUrl="${baseUrlSourceGitlabApi}/groups/${groupPathEncoded}"
 
   # https://docs.gitlab.com/ee/api/groups.html#list-a-groups-projects
-  # TODO do we need to follow paging or ist it safe to assume that no group has more than 100 projects?
   local groupProjectsUrl="${groupsUrl}/projects?per_page=100&simple=true"
+
   local -a projects
-  mapfile -t projects < <(curl -sS --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}" | jq -r '.[].path_with_namespace')
+  mapfile -t projects <<< "$(getObjects)"
+
   local -a archivedProjects
-  mapfile -t archivedProjects < <(curl -sS --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}&archived=true" | jq -r '.[].path_with_namespace')
+  mapfile -t archivedProjects <<< "$(getObjects "archived")"
+
+  if [[ "${#archivedProjects[@]}" == 1 ]]; then
+    archivedProjects=()
```

Comment:
Could you elaborate why this condition exists?

Reply:
Oh, I'm sorry for this kludge. I get empty array without it, if I don't have archived projects. Example output without it:

```
Migrating project '': <-- sic!
	Skipping already existing target project.
```

I haven't thought of anything better :(

Reply:
Thanks, I'll have a look into this once I find the time. There should be a more clean solution for your issue.